presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -130,8 +133,19 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Context> c
         {
             context.get().variables.addAll(node.getGroupingKeys());
 
-            for (AggregationNode.Aggregation aggregation : node.getAggregations().values()) {
-                aggregation.getArguments().forEach(expression -> subfieldExtractor.process(castToExpression(expression), context.get()));
+            for (Map.Entry<VariableReferenceExpression, AggregationNode.Aggregation> entry : node.getAggregations().entrySet()) {
+                VariableReferenceExpression variable = entry.getKey();
+                AggregationNode.Aggregation aggregation = entry.getValue();
+
+                // Aggregate calls with pass-through functions should allow sub-field pruning
```

Comment:
nit: `Allow sub-field pruning to pass though arbitrary() aggregation`

Reply:
Done